packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts

```diff
@@ -151,97 +187,159 @@ export abstract class MikroOrmAbstractBaseRepository<T = any>
-
-  abstract create(data: unknown, context?: Context): Promise<T>
-
-  abstract delete(id: string, context?: Context): Promise<void>
-}
-
-/**
- * Privileged extends of the abstract classes unless most of the methods can't be implemented
- * in your repository. This base repository is also used to provide a base repository
- * injection if needed to be able to use the common methods without being related to an entity.
- * In this case, none of the method will be implemented except the manager and transaction
- * related ones.
- */
-
-export class MikroOrmBaseRepository extends MikroOrmAbstractBaseRepository {
-  constructor({ manager }) {
-    // @ts-ignore
-    super(...arguments)
-  }
-
-  create(data: unknown[], context?: Context): Promise<any[]> {
+  ): Promise<[T[], number]> {
     throw new Error("Method not implemented.")
   }
 
-  update(data: unknown[], context?: Context): Promise<any[]> {
+  create(data: unknown, context?: Context): Promise<T> {
     throw new Error("Method not implemented.")
   }
 
-  delete(ids: string[], context?: Context): Promise<void> {
+  delete(id: string, context?: Context): Promise<void> {
     throw new Error("Method not implemented.")
   }
+}
 
-  find(options?: DAL.FindOptions, context?: Context): Promise<any[]> {
-    throw new Error("Method not implemented.")
-  }
+type DtoBasedMutationMethods = "create" | "update"
 
-  findAndCount(
-    options?: DAL.FindOptions,
-    context?: Context
-  ): Promise<[any[], number]> {
-    throw new Error("Method not implemented.")
+export function mikroOrmBaseRepositoryFactory<
+  T extends object = object,
+  TDTos extends { [K in DtoBasedMutationMethods]?: any } = {
+    [K in DtoBasedMutationMethods]?: any
   }
-}
+>(
+  entity: EntityClass<T> | EntitySchema<T> | string,
+  primaryKey: string = "id"
+) {
+  class MikroOrmAbstractBaseRepository_ extends MikroOrmBaseRepository<T> {
+    async create(data: TDTos["create"][], context?: Context): Promise<T[]> {
+      const manager = this.getActiveManager<EntityManager>(context)
+
+      const entities = data.map((data_) => {
+        return manager.create(
+          entity as EntityName<T>,
+          data_ as RequiredEntityData<T>
+        )
+      })
+
+      manager.persist(entities)
+
+      return entities
+    }
 
-export class MikroOrmBaseTreeRepository extends MikroOrmAbstractTreeRepositoryBase {
-  constructor({ manager }) {
-    // @ts-ignore
-    super(...arguments)
-  }
+    async update(data: TDTos["update"][], context?: Context): Promise<T[]> {
+      const manager = this.getActiveManager<EntityManager>(context)
 
-  find(
-    options?: DAL.FindOptions,
-    transformOptions?: RepositoryTransformOptions,
-    context?: Context
-  ): Promise<any[]> {
-    throw new Error("Method not implemented.")
-  }
+      const primaryKeyValues: string[] = data.map((data_) => data_[primaryKey])
+      const existingEntities = await this.find(
+        {
+          where: {
+            [primaryKey]: {
```

Comment:
It doesn't need to be in this PR, but eventually we'll need to support composite primary keys to update and delete.

Reply:
Yes I thought about it too, but I was more in favor to have it in a separate pr as soon as we need that. knowing that this one is the base class it can be overridden for now